index.ts

```diff
@@ -1,110 +1,110 @@
+        : Promise<youtube_v3.Schema$PlaylistItem[]> {
+    const res = await getVideos(nextPageToken);
+    if (!res) return [];
+    const newItems = items.concat(res.videos);
     if (count < 1) {
         // if (count < res.totalResults / 50) {
-        return getAllVideos(res.nextPageToken = "", newItems, count + 1)
+        return getAllVideos(res.nextPageToken = '', newItems, count + 1);
     } else {
-        return newItems
+        return newItems;
     }
 }
 
-function getRating(review): string {
-    if (review.includes("CLASSIC")) return "CLASSIC/10"
-    if (review.includes("NOT GOOD")) return "NOT GOOD/10"
+function getRating(review: string): string {
+    if (review.includes('CLASSIC')) return 'CLASSIC/10';
+    if (review.includes('NOT GOOD')) return 'NOT GOOD/10';
 
     const regex = /[0-9][10]*\/[1][0]/g;
-    const rating = review.match(regex) || "??/??";
+    const rating = review.match(regex) || '??/??';
     return rating[0];
 }
 
-function getArtist(review): string {
-    if (review.resourceId.videoId === "F-Fd5YG2pWs") return
-    if (review.resourceId.videoId === "MNnibsPJSDY") return
-    if (review.resourceId.videoId === "LDMNhCOs0G0") return
-
-    const regex = /.*(?=-)/g;
-    const artist = review.title.trim().match(regex);
-    return artist[0].replace(/\s*$/, "");
+function getArtist(review: youtube_v3.Schema$PlaylistItemSnippet): string {
+    return getRegexGroupFromTitle(review, /.*(?=-)/g);
 }
 
-function getAlbum(review): string {
-    if (review.resourceId.videoId === "F-Fd5YG2pWs") return
-    if (review.resourceId.videoId === "MNnibsPJSDY") return
-    if (review.resourceId.videoId === "LDMNhCOs0G0") return
-
-    const regex = /(?<=-).*/g;
-    const album = review.title.trim().match(regex)
-    // .replace("ALBUM REVIEW", "")
-    return album[0].replace(/\s*$/, "");
+function getAlbum(review: youtube_v3.Schema$PlaylistItemSnippet): string {
+    return getRegexGroupFromTitle(review, /(?<=-).*/g);
 }
 
-interface Review {
-    date: string,
-    url: string,
-    artist: string,
-    album: string,
-    rating: string
+function isBlacklistedReview(review: youtube_v3.Schema$PlaylistItemSnippet) : boolean {
+    if (!review || !review?.resourceId?.videoId) return true;
+    return ['F-Fd5YG2pWs', 'MNnibsPJSDY', 'LDMNhCOs0G0'].includes(review.resourceId.videoId);
 }
 
-(async function () {
-    const allVideos = await Promise.resolve(getAllVideos("", [], 0));
-    const albumReviews = allVideos.flatMap(v => v.snippet.title.endsWith("ALBUM REVIEW") ?
-        [v.snippet] : [])
-
-    const snippets = albumReviews.map(review => ({
-        date: review.publishedAt.substring(0, 10),
-        url: `youtube.com/watch?v=${review.resourceId.videoId}`,
-        artist: getArtist(review),
-        album: getAlbum(review),
-        rating: getRating(review.description)
-    }))
+function isPresent<T>(t: T | undefined | null | void): t is T {
+    return t !== undefined && t !== null;
```

Comment:
Is an empty string or zero considered "Present"?
```suggestion
    return !!t;
```

Reply:
Aye, was too narrow in scope.